### ISSUE TITLE
Changing the cluster identifier in an aws_rds_cluster resource should not destroy/recreate the cluster

### ISSUE THREAD

**Issue by gblues** [hashicorp/terraform-provider-aws#20363]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform version: `1.0.3`
AWS provider version: `3.42.0`

### Affected Resource(s)

* aws_rds_cluster
* aws_rds_cluster_instance

### Terraform Configuration Files

NOTE: Creating the networking infrastructure (VPC, subnets, and db subnet group) are left as an exercise to the reader since they are not relevant to the issue.

```hcl
locals {
  instance_identifier = "my-name"
  cluster_identifier = local.instance_identifier
  engine_version = "5.7.mysql_aurora.2.04.5"
}

resource "aws_rds_cluster" "aurora" {
  cluster_identifier = local.cluster_identifier
  engine             = "aurora-mysql"
  engine_version     = local.engine_version
  port               = 3306
  database_name      = "te"
  master_username    = "master"
  apply_immediately  = true
  storage_encrypted  = false

  master_password      = "choose_your_own_password"
  db_subnet_group_name = "your-db-subnet-group-name"


  deletion_protection       = false
  skip_final_snapshot       = true

  vpc_security_group_ids          = []
  backup_retention_period         = 1

  enabled_cloudwatch_logs_exports = []
}

resource "aws_rds_cluster_instance" "aurora" {
  cluster_identifier              = local.cluster_identifier
  identifier                      = local.instance_identifier
  instance_class                  = "db.t3.medium"
  engine                          = "aurora-mysql"
  engine_version                  = local.engine_version
  ca_cert_identifier              = "rds-ca-2019"
  depends_on                      = [aws_rds_cluster.aurora]
  apply_immediately               = true
  auto_minor_version_upgrade      = false
  performance_insights_enabled    = false
}
```

### Debug Output

https://gist.github.com/gblues/082f3fac11b5a6bc72c717f25493a94b

### Expected Behavior

The cluster should be renamed.

### Actual Behavior

The cluster is destroyed and re-created, meaning any data stored in the cluster is lost.

### Steps to Reproduce

1. Apply the above code
2. Change the cluster identifier to `"${local.instance_identifier}-cluster"`
3. run `terraform plan`

### Important Factoids
By contrast, the AWS console allows one to modify the cluster and change the cluster ID with no data loss.

### References

I found an issue opened years ago on this issue on the main terraform repo, but it was never interacted with:

https://github.com/hashicorp/terraform/issues/11636

**Comment by anGie44** [comment]
Referenced in https://github.com/hashicorp/terraform-provider-aws/issues/507

**Comment by gblues** [comment]
Appears to be resolved in the 5.0.0 release of the AWS provider. Thank you!

https://github.com/hashicorp/terraform-provider-aws/issues/507#issuecomment-1563262964

**Comment by enchorb** [comment]
still not solved for `aws_rds_cluster_instance`
